File: springboot-backend/src/main/java/com/capstone/project/service/NotificationService.java
package com.capstone.project.service;

import com.capstone.project.exception.ResourceNotFroundException;
import com.capstone.project.model.Class;
import com.capstone.project.model.Notification;

import java.text.ParseException;
import java.util.List;
import java.util.Map;

public interface NotificationService {
    List<Notification> getAllNotification();

    Notification getNotificationById(int id) throws ResourceNotFroundException;

    Notification getNotificationByUserId(int id) throws ResourceNotFroundException;

    Notification createNotification(Notification noti);

    Notification updateNotification( Notification noti,  int id) throws ResourceNotFroundException;
    Boolean deleteNotification( int id) throws ResourceNotFroundException;

    Map<String, Object> getFilterNotification(String content, Boolean isRead, String title, String fromdatetime, String todatetime, int userid, String direction, int page, int size);
}

File: react-frontend/src/services/NotificationService.js
import axios from 'axios'

const API_BASE_URL = process.env.REACT_APP_HOST

const createNotification = (notification) => {
    return axios.post(API_BASE_URL + '/notification', notification)
}

const updateNotification = (id, notification) => {
    return axios.put(API_BASE_URL + '/notification/' + id, notification)
}

const markAsRead = (id) => {
    return axios.get(API_BASE_URL + '/markasread/' + id)
}

const getFilterNotification = (
    content,
    isRead,
    title,
    fromDatetime,
    toDatetime,
    userid,
    direction,
    page,
    size
) => {
    return axios.get(
        API_BASE_URL +
            '/filternotification?content' +
            content +
            '&isread' +
            isRead +
            '&title' +
            title +
            '&fromdatetime' +
            fromDatetime +
            '&todatetime' +
            toDatetime +
            '&userid' +
            userid +
            '&direction' +
            direction +
            '&page' +
            page +
            '&size' +
            size
    )
}

const NotificationService = {
    createNotification,
    getFilterNotification,
    updateNotification,
    markAsRead,
}

export default NotificationService
